grafast/grafserv/__tests__/errors.test.ts

```diff
@@ -0,0 +1,40 @@
+import { fetch } from "@whatwg-node/fetch";
+
+import { makeExampleServer } from "./exampleServer.js";
+
+let server: Awaited<ReturnType<typeof makeExampleServer>> | null = null;
+
+afterAll(() => {
+  server?.release();
+});
+
+test("response body contains expected error object when function provided as grafast context option throws an error", async () => {
+  const maskError = jest.fn((error) => error);
+  server = await makeExampleServer({
+    grafserv: {
+      graphqlOverGET: true,
+      graphqlPath: "/graphql",
+      dangerouslyAllowAllCORSRequests: true,
+      maskError,
+    },
+    grafast: {
+      context: () => {
+        throw new Error("a particular error");
+      },
```

Comment:
If you're looking to get another PR merged... Adding another test where context sleeps 5ms and then throws an error would be good - to differentiate a synchronous throw vs an async reject.